packages/integrations/mdx/src/vite-plugin-mdx.ts

```diff
@@ -44,9 +44,10 @@ export function vitePluginMdx(mdxOptions: MdxOptions): Plugin {
 		async transform(code, id) {
 			if (!id.endsWith('.mdx')) return;
 
-			const { data: frontmatter, content: pageContent } = parseFrontmatter(code, id);
+			const { data: frontmatter, content: pageContent, matter } = parseFrontmatter(code, id);
+			const frontmatterLines = matter ? matter.match(/\n/g)?.join('') + '\n\n' : '';
 
-			const vfile = new VFile({ value: pageContent, path: id });
+			const vfile = new VFile({ value: frontmatterLines + pageContent, path: id });
```

Comment:
Is there any risk of this being breaking?

Reply:
I don't think so, it's only prepending new lines. Do you have something in mind that could break?

Reply:
I figured that perhaps a remark/hype plugin could be relying on certain positions, but I guess that's unlikely

Reply:
Hmm I guess you're right since the mdast/hast nodes do have offset information, but since they only work upon the `vfile.value`, it should still be fine. Just that if they return any offset values then it'd be incorrect, but I don't think you'd do so in practice, except in the linked issue where the returned error uses line/col instead of offsets.

We could replace the frontmatter with spaces and new lines, but I think maybe for now new lines could be enough and we can monitor it again.